server/processors-alerts-publisher/src/main/java/org/hawkular/apm/processor/alerts/TraceCompletionAlertsPublisherMDB.java

```diff
@@ -57,9 +59,37 @@ public void onMessage(Message message) {
         try {
             String data = ((TextMessage) message).getText();
             List<CompletionTime> items = mapper.readValue(data, new TypeReference<List<CompletionTime>>() {});
-            publisher.publish(items, "TraceCompletion");
+            List<Event> events = items.stream()
```

Comment:
Same as the previous comment: instead of building a `List`, it might be best to do a `forEach` and call `#publish(Event)` directly.